### ISSUE TITLE
parse() keeps returning the same data even if you call it with new HTML

### ISSUE THREAD

**Issue by callumlocke** [Swaagie/minimize#65]
```js
var minimize = new Minimize();

minimize.parse(html, function (err, minifiedHtml) {
 //
});
```

The above works fine, but if I later call `minimize.parse()` again, but pass it a completely different string HTML, it gives me back the same response as last time.

Is that intentional?

**Comment by Swaagie** [comment]
hmm no that sounds like a bug, but I don't think this is caused by minimize. For example, I did

```js
var minimize = new (require('minimize'))();
minimize.parse('<h1 class=>something</h1>', function (e, m) { console.log(m); })
// <h1>something</h1>
minimize.parse('<h1 class=>different</h1>', function (e, m) { console.log(m); })
// <h1>different</h1>
```
and it seems to work fine, is there something else in your code that could cause this?

**Comment by callumlocke** [comment]
No it's definitely a bug in Minimize. Here's a demo:
https://tonicdev.com/callumlocke/minimize-bug-demo

It works OK if you wait for one parse() call to finish before calling it again with new HTML:
https://tonicdev.com/callumlocke/minimize-bug-workaround

Maybe in your test you manually entered the lines one at a time in your console? If so, that would work OK. The problem is if you're minifying a bunch of HTML files in parallel, it doesn't work.

It looks like the gulp plugin manually [works around this bug](https://github.com/murphydanger/gulp-minify-html/blob/ed969a55c335e6760cc6623b2e68fb4c0c37e428/index.js#L16-L25) by making a new Mimimize instance for every parse() call.

**Comment by Swaagie** [comment]
Hmm k that makes sense, definitly a bug, the culprit would be `https://github.com/Swaagie/minimize/blob/master/lib/minimize.js#L64`. That would need to be a unique event per `parse` call

**Comment by Swaagie** [comment]
released a fix with 1.7.4, thanks for reporting :) have a good one
